crates/codegen/runtime/cargo/crate/src/ir/builder.rs.jinja2

```diff
@@ -181,32 +170,49 @@
       self.index < self.children.len() && self.children[self.index].label == label
     }
 
-    fn accept_label(&mut self, label: EdgeLabel) -> Result<&Node> {
+    fn accept_label(&mut self, label: EdgeLabel) -> Option<&Node> {
       if self.index >= self.children.len() {
-        return Err(format!("Expected more sibling nodes, looking for label {label:?}"));
+        // we may end up in this case if the parser generates an incomplete CST
+        // node because of a parser error
+        return None;
       }
-      if self.children[self.index].label == label {
-        let node = &self.children[self.index].node;
+      if !self.children[self.index].is_valid() {
+        // attempt to recover from an unrecognized/missing node
         loop {
           self.index += 1;
-          if self.index >= self.children.len() || !self.children[self.index].is_trivia() {
+          if self.index >= self.children.len() {
+            return None;
+          }
+          if !self.children[self.index].is_trivia() {
             break;
           }
         }
-        Ok(node)
-      } else {
-        Err(format!("Expected node with label {label:?}, got {:?}", self.children[self.index].label))
+        if self.children[self.index].label != label {
+          return None;
+        }
       }
+      assert!(
+        self.children[self.index].label == label,
+        "expected node with label {label}, instead got {edge:?}", edge = self.children[self.index],
+      );
+
+      let node = &self.children[self.index].node;
+      loop {
+        self.index += 1;
+        if self.index >= self.children.len() || !self.children[self.index].is_trivia() {
```

Comment:
we already skip leading trivia in each loop above, and also trailing trivia in `finalize()` below.
I wonder why is the added check here? when is it needed?

Reply:
This is now removed.